U/ejeffrey/fasttypes

ejeffrey opened the pull request:
This fixes the crash when no type tag matches and adds tests.

maffoo commented on labrad/test/test_fasttypes.py:
```diff
@@ -8,14 +8,20 @@ def test_int():
+    data, tt = str(data), str(tt)
+    assert all(ft.unflatten((data, tt)) == [1,2,3])
+    assert len(ft.flatten(['a', 'bc', 'def'])[0])==22
+def test_list_w_tag():
```
style: add blank lines between top-level definitions (https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Blank_Lines#Blank_Lines)

maffoo commented on labrad/test/test_fasttypes.py:
```diff
@@ -8,14 +8,20 @@ def test_int():
     assert all(T.unflatten(*ft.flatten([1,2,3])) == [1,2,3])
-    assert all(ft.unflatten(T.flatten([1,2,3])) == [1,2,3])
+    (data, tt) = T.flatten([1,2,3])
+    data, tt = str(data), str(tt)
```
why do you have to call `str` on these? what is the actual return type?

ejeffrey replied:
fasttypes only works with strings
types.py returns LRType objects and sometimes uses Buffer()'s for the data.

On Thu, Apr 23, 2015 at 2:11 PM, Matthew Neeley <notifications@github.com>
wrote:

> In labrad/test/test_fasttypes.py
> <https://github.com/martinisgroup/pylabrad/pull/91#discussion_r29003971>:
>
> >      with pytest.raises(Exception):
> >          ft.flatten('abcd', 'i')
> >      with pytest.raises(Exception):
> >          ft.flatten(4, 's')
> >  def test_list():
> >      assert all(T.unflatten(*ft.flatten([1,2,3])) == [1,2,3])
> > -    assert all(ft.unflatten(T.flatten([1,2,3])) == [1,2,3])
> > +    (data, tt) = T.flatten([1,2,3])
> > +    data, tt = str(data), str(tt)
>
> why do you have to call str on these? what is the actual return type?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/martinisgroup/pylabrad/pull/91/files#r29003971>.
>

maffoo replied:
ah, right, these are coming from `types.py` not `fasttypes`

maffoo commented on labrad/test/test_fasttypes.py:
```diff
@@ -8,14 +8,20 @@ def test_int():
+    (data, tt) = T.flatten([1,2,3])
+    data, tt = str(data), str(tt)
+    assert all(ft.unflatten((data, tt)) == [1,2,3])
+    assert len(ft.flatten(['a', 'bc', 'def'])[0])==22
```
nit: whitespace around operators

maffoo commented:
LGTM

ejeffrey commented:
Fixed a memory leak austin pointed out.  I also found a few more memory
leaks and uninitialized memory use with valgrind that still need to be
fixed.  I want to get him to review this. @afowler.

On Thu, Apr 23, 2015 at 2:38 PM, Matthew Neeley <notifications@github.com>
wrote:

> LGTM
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95727111>.
>

afowler commented:
Would love to review the changes, struggling to find them...

On Thu, Apr 23, 2015 at 2:50 PM, ejeffrey <notifications@github.com> wrote:

> Fixed a memory leak austin pointed out. I also found a few more memory
> leaks and uninitialized memory use with valgrind that still need to be
> fixed. I want to get him to review this. @afowler.
>
> On Thu, Apr 23, 2015 at 2:38 PM, Matthew Neeley <notifications@github.com>
> wrote:
>
> > LGTM
> >
> > --
> > Reply to this email directly or view it on GitHub
> > <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95727111
> >.
>
> >
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95728952>.
>

ejeffrey commented:
If you follow the github link at the bottom of the email (Reply to this
email directly or view it on GitHub) you can see the history of comments on
the pull request.  Near the top there are 'tabs' for
Conversation, Commits, and Files changed.  Files changed shows the total
difference between my version and yours, while commits lets you look at
invidual commits.


On Thu, Apr 23, 2015 at 3:03 PM, afowler <notifications@github.com> wrote:

> Would love to review the changes, struggling to find them...
>
> On Thu, Apr 23, 2015 at 2:50 PM, ejeffrey <notifications@github.com>
> wrote:
>
> > Fixed a memory leak austin pointed out. I also found a few more memory
> > leaks and uninitialized memory use with valgrind that still need to be
> > fixed. I want to get him to review this. @afowler.
> >
> > On Thu, Apr 23, 2015 at 2:38 PM, Matthew Neeley <
> notifications@github.com>
> > wrote:
> >
> > > LGTM
> > >
> > > --
> > > Reply to this email directly or view it on GitHub
> > > <
> https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95727111
> > >.
> >
> > >
> >
> > --
> > Reply to this email directly or view it on GitHub
> > <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95728952
> >.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95731670>.
>

afowler commented:
Hi Evan,

So these extra memory issues have not been addressed yet?

> > I also found a few more memory
> > leaks and uninitialized memory use with valgrind that still need to be
> > fixed.

Assuming this is the case, LGTM.

On Thu, Apr 23, 2015 at 3:08 PM, ejeffrey <notifications@github.com> wrote:

> If you follow the github link at the bottom of the email (Reply to this
> email directly or view it on GitHub) you can see the history of comments on
> the pull request. Near the top there are 'tabs' for
> Conversation, Commits, and Files changed. Files changed shows the total
> difference between my version and yours, while commits lets you look at
> invidual commits.
>
>
> On Thu, Apr 23, 2015 at 3:03 PM, afowler <notifications@github.com> wrote:
>
> > Would love to review the changes, struggling to find them...
> >
> > On Thu, Apr 23, 2015 at 2:50 PM, ejeffrey <notifications@github.com>
> > wrote:
> >
> > > Fixed a memory leak austin pointed out. I also found a few more memory
> > > leaks and uninitialized memory use with valgrind that still need to be
> > > fixed. I want to get him to review this. @afowler.
> > >
> > > On Thu, Apr 23, 2015 at 2:38 PM, Matthew Neeley <
> > notifications@github.com>
> > > wrote:
> > >
> > > > LGTM
> > > >
> > > > --
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95727111
> > > >.
> > >
> > > >
> > >
> > > --
> > > Reply to this email directly or view it on GitHub
> > > <
> https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95728952
> > >.
> >
> > >
> >
> > --
> > Reply to this email directly or view it on GitHub
> > <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95731670
> >.
>
> >
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/martinisgroup/pylabrad/pull/91#issuecomment-95732362>.
>
